src/Client/ClientBase.cpp

```diff
@@ -1746,6 +1746,21 @@ bool isStdinNotEmptyAndValid(ReadBuffer & std_in)
 {
     try
     {
+        // Use non-blocking check for stdin to avoid hanging
+        if (auto * fd_buffer = typeid_cast<ReadBufferFromFileDescriptor *>(&std_in))
+        {
+            int fd = fd_buffer->getFD();
+            if (fd == STDIN_FILENO)
+            {
+                int flags = fcntl(fd, F_GETFL);
+                if (flags != -1)
+                {
+                    fcntl(fd, F_SETFL, flags | O_NONBLOCK);
```

Comment:
This changed the behavior of stdin empty check. The orginal one will block wait for data if not eol found.
I still think this mod is risky while processing large chunk of input data with slow IO.